### PR TITLE
feat(ruby): add CTS

### PR DESCRIPTION
**Pull request by millotp**
## 🧭 What and Why

🎟 JIRA Ticket: [DI-1728](https://algolia.atlassian.net/browse/DI-1728)

Generate the CTS for ruby, using the standard unit test.

[DI-1728]: https://algolia.atlassian.net/browse/DI-1728?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: tests/output/ruby/Gemfile.lock
Comment (shortcuts) [inline tests/output/ruby/Gemfile.lock:1]: maybe this needs to be generated/pushed by the CI? Just so the version updates

**Review by shortcuts — COMMENTED** [review]
nothing to add it looks really clean

File: tests/output/ruby/Gemfile.lock
Comment (millotp) [reply tests/output/ruby/Gemfile.lock:1]: ah yes good idea

File: scripts/ci/githubActions/createMatrix.ts [hunk 1]
```diff
@@ -122,6 +122,9 @@ async function createClientMatrix(baseBranch: string): Promise<void> {
 
         testsToStore = `${testsToStore} ${testsRootFolder}/package.json`;
         break;
+      case 'ruby':
```
Comment (shortcuts) [inline scripts/ci/githubActions/createMatrix.ts:125]: I forgot it for Python, could you please store the `poetry.lock` file? :D

File: templates/ruby/api.mustache [hunk 2]
```diff
@@ -44,30 +43,18 @@ module {{moduleName}}
     {{#notes}}
     # {{{.}}}
     {{/notes}}
-{{#vendorExtensions.x-group-parameters}}
-    # @param [Hash] opts the parameters
 {{#allParams}}
 {{#required}}
-    # @option opts [{{{dataType}}}] :{{paramName}} {{description}} (required)
+    # @param {{paramName}} [{{{dataType}}}] {{description}} (required)
```
Comment (shortcuts) [inline templates/ruby/api.mustache:48]: it's a param when it's required and option when it's optional?

File: templates/ruby/api.mustache [hunk 3]
```diff
@@ -77,39 +64,17 @@ module {{moduleName}}
...
-      {{^isNullable}}
-      {{#required}}
-      {{{paramName}}} = opts[:{{{paramName}}}]
-      {{/required}}
-      {{/isNullable}}
-      {{/allParams}}
-      {{/vendorExtensions.x-group-parameters}}
+    # @option request_options: The request options to send along with the query, they will be merged with the transporter base parameters (headers, query params, timeouts, etc.). (optional)
+    # @return [Array<({{{returnType}}}{{^returnType}}nil{{/returnType}}, Http::Response)>] {{#returnType}}{{{.}}} data{{/returnType}}{{^returnType}}nil{{/returnType}} and response
+    def {{operationId}}_with_http_info({{#allParams}}{{#required}}{{paramName}}, {{/required}}{{^required}}{{paramName}}=nil, {{/required}}{{/allParams}}request_options = {})
```
Comment (shortcuts) [inline templates/ruby/api.mustache:77]: wdyt of defining the with_http_info one before the deserializing one? :D

**Review by shortcuts — COMMENTED** [review]
so nice

File: templates/ruby/api.mustache (hunk 2 above)
Comment (millotp) [reply templates/ruby/api.mustache:48]: maybe not, they should all be `@param`, and `@option` to describe what goes into `request_options` for example

File: templates/ruby/api.mustache (hunk 3 above)
Comment (millotp) [reply templates/ruby/api.mustache:77]: if you want ahah

**Review by shortcuts — APPROVED** [review]
AWESOME

**Pull request merged**